move-to-sdk/app/io/sphere/sdk/facets/RangeBucketFacetImpl.java

```diff
@@ -0,0 +1,115 @@
+package io.sphere.sdk.facets;
+
+import io.sphere.sdk.search.FacetedSearchExpression;
+import io.sphere.sdk.search.PagedSearchResult;
+import io.sphere.sdk.search.RangeFacetResult;
+import io.sphere.sdk.search.model.*;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+public class RangeBucketFacetImpl<T> extends BaseFacet<T> implements RangeBucketFacet<T> {
+
+    private final List<FacetRange> selectedRanges;
+    @Nullable private final RangeFacetResult facetResult;
+    private final List<RangeBucketOption> facetOptions;
+    private final RangeTermFacetedSearchSearchModel rangeTermFacetedSearchSearchModel;
+
+    protected RangeBucketFacetImpl(final String key, final String label, final boolean countHidden, final FacetType type,
+                             final FacetedSearchSearchModel<T> searchModel, final List<FacetRange> selectedRanges,
+                             @Nullable final RangeFacetResult facetResult, RangeTermFacetedSearchSearchModel rangeTermFacetedSearchSearchModel) {
+        super(key, type, countHidden, label, searchModel);
+        this.selectedRanges = selectedRanges;
+        this.facetResult = facetResult;
+        this.facetOptions = initializeOptions(selectedRanges, facetResult);
+        this.rangeTermFacetedSearchSearchModel = rangeTermFacetedSearchSearchModel;
+    }
+
+    @Override
+    public List<FacetRange> getSelectedRanges() {
+        return selectedRanges;
+    }
+
+    @Nullable
+    @Override
+    public RangeFacetResult getFacetResult() {
+        return facetResult;
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return facetOptions.size()>1;
+    }
+
+    @Override
+    public List<RangeBucketOption> getFacetOptions() {
+        return facetOptions;
+    }
+
+    @Override
+    public List<String> getSelectedValues() {
+        return null;
+    }
+
+    @Override
+    public FacetedSearchExpression<T> getFacetedSearchExpression() {
+        final FacetedSearchExpression<T> facetedSearchExpr;
+        if (selectedRanges.isEmpty()) {
+            facetedSearchExpr = rangeTermFacetedSearchSearchModel.allRanges();
+        } else {
+            facetedSearchExpr = rangeTermFacetedSearchSearchModel.isBetweenAny(
+                    selectedRanges.stream()
+                        .map(facetRange -> FilterRange.of(facetRange.lowerEndpoint().toString(), facetRange.upperEndpoint().toString()))
+                        .collect(toList())
```

Comment:
Actually `FilterRange` is the one to use here (as you can notice :P). It has some complexity which I'll try to explain here but I'm always up for a call:
- Filter/FacetRange is internally the same but the allowed ranges are different (that's why we have 2 classes instead of just Range).
- The FacetedSearch model accepts only `FilterRange` because it's NOT well prepared for bucket ranges (that's why we needed to implement this and try). It is assuming you'll always want an initial range of `(* to *)` and thus it only needs to know what are you going to filter by (the filter expression), which is what the `SliderRangeFacet` needs. But the bucket needs to additionally define the initial range (the facet expression). To solve that problem, for now you'll have to create a new `RangeFacetedSearchExpression`, providing the facet and filter expressions yourself:
```
@Override
public FacetedSearchExpression<T> getFacetedSearchExpression() {
    final FacetedSearchExpression<T> facetedSearchExpr;
    if (selectedRanges.isEmpty()) {
        facetedSearchExpr = rangeTermFacetedSearchSearchModel.allRanges();
    } else {
        facetedSearchExpr = rangeTermFacetedSearchSearchModel.isBetweenAny(selectedRanges);
    }
    return buildBucketRangeFacetedSearchExpression(facetedSearchExpr);
}

private RangeFacetedSearchExpression<T> buildBucketRangeFacetedSearchExpression(final FacetedSearchExpression<T> facetedSearchExpr) {
    final List<FacetRange<String>> facetRanges = Collections.emptyList(); // TODO this is an example of facet ranges from bucket range, should be defined in the class
    final RangeTermFacetSearchModel<T, String> searchModel = extractSearchModel(facetedSearchExpr);
    return RangeFacetedSearchExpression.of(searchModel.onlyRange(facetRanges), facetedSearchExpr.filterExpressions());
}

private RangeTermFacetSearchModel<T, String> extractSearchModel(final FacetedSearchExpression<T> facetedSearchExpr) {
    final String attributePath = facetedSearchExpr.facetExpression().attributePath();
    final RangeTermFacetSearchModel<T, String> searchModel = RangeTermFacetSearchModel.of(attributePath, TypeSerializer.ofString());
    return searchModel.withAlias(facetedSearchExpr.facetExpression().alias());
}
```
So in the case of Bucket range facet, you'll need an additional `List<FacetRange>`.